chromium_src/components/content_settings/core/common/cookie_settings_base.cc

```diff
@@ -0,0 +1,107 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/content_settings/core/common/cookie_settings_base.h"
+
+#include "base/containers/flat_map.h"
+#include "base/no_destructor.h"
+#include "base/optional.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+#include "components/content_settings/core/common/content_settings.h"
+#include "components/content_settings/core/common/content_settings_pattern.h"
+#include "url/gurl.h"
+#include "url/origin.h"
+
+namespace content_settings {
+
+namespace {
+
+constexpr char kWp[] = "https://[*.]wp.com/*";
+constexpr char kWordpress[] = "https://[*.]wordpress.com/*";
+constexpr char kPlaystation[] = "https://[*.]playstation.com/*";
+constexpr char kSonyentertainmentnetwork[] =
+    "https://[*.]sonyentertainmentnetwork.com/*";
+
+bool BraveIsAllowedThirdParty(
+    const GURL& url,
+    const GURL& site_for_cookies,
+    const base::Optional<url::Origin>& top_frame_origin) {
+  static const base::NoDestructor<
+      // url -> first_party_url allow map
+      std::vector<std::pair<ContentSettingsPattern,
+                            ContentSettingsPattern>>> entity_list({
+          {
+            ContentSettingsPattern::FromString(kWp),
```

Comment:
Instead of hard-coding the exceptions in the code, we can use the local-data-updater component to create a cookie parsing list similar to: https://github.com/brave/referrer-whitelist/blob/master/data/ReferrerWhitelist.json

cc: @diracdeltas

Reply:
I'm pretty sure that eventually we will end up with a server-updated list of "entities" (something like https://github.com/krgovind/first-party-sets or another similar attempt)

Reply:
First party sets are not getting consensus from all browser vendors. There are too many edge-cases.

Reply:
Strong vote against full party sets from me too.  But would be open to pulling in user-agent controlled set of site associations (a la disconnect entities list). 

I dig the local-data-updater idea too, but for my two cents, i'd like to separate that from this PR, since a this PR solves web compat issues folks are complaining about now, and local-data-updater would take a while longer (and, this machinery would more or less be needed anyway for the local-data-updater approach, so this can be one step on the way to that)

Reply:
I don't think `local-data-updater` idea will take much time. We can use the `referrer-exceptions` list for cookies. I'm happy to quickly set-up a separate list if there are any concerns with using the `referrer whitelist`

Reply:
I think a separate list would be better.  @bridiver would probably know best whether changing this to `local-data-updater` is better to do in this PR, or in some future one.  

For my two cents, if it pushes landing this PR back more than a day or two, its best doing separately though

Reply:
yeah I also think we can make a separate list separately

Reply:
Again, no argument about it going in a separate list, thats fine.  But time is sensitive here, for biz reasons.  If it will take longer to make a sep list, lets please merge this, and then work on a separate list as a next time.

Reply:
@pes10k and I already discussed a list as a downloadable component, but that is outside the scope of this PR. We are only adding one exception to the existing hardcoded list from https://github.com/brave/brave-core/pull/5433/files#diff-9981585c3d9184fc54b96697c59d2028L332

Reply:
and that exception was just added to this PR because it was somewhat related and we were moving the existing exception